src/diffusers/models/unet_2d_condition.py

```diff
@@ -123,20 +124,45 @@ def __init__(
         num_class_embeds: Optional[int] = None,
         upcast_attention: bool = False,
         resnet_time_scale_shift: str = "default",
+        time_embed_dim: int = None,
```

Comment:
would be great to be able to remove this parameter and instead derive it automatically when`"time_embedding_str"= "fourier"` is used.